spec/models/sources/linode_spec.rb

```diff
@@ -0,0 +1,182 @@
+RSpec.describe Sources::Linode, type: :model do
+  let(:source) {create(:linode_source)}
+  let(:site) {create(:site)}
+  let(:proxy) {create(:proxy)}
+  let(:proxy_performance) do
+    create(:proxy_performance, :proxy => proxy, :site => site)
+  end
+
+  context '#validate_config!' do
+    it 'identifies a valid config' do
+      expect(source).to receive(:names_to_ids)
+      expect(source).to receive(:image_id).and_return 4
+      expect(source).to receive(:flavor_id).and_return 5
+      expect(source).to receive(:kernel_id).and_return 6
+      expect(source).to receive(:data_center_id).and_return 7
+
+      expect(source.validate_config!).to be_truthy
+    end
+
+    it 'identifies an invalid config when an id is nil' do
+      expect(source).to receive(:names_to_ids)
+      expect(source).to receive(:image_id).and_return 4
+      expect(source).to receive(:flavor_id).and_return 5
+      expect(source).to receive(:kernel_id).and_return 6
+      expect(source).to receive(:data_center_id).and_return nil
+
+      expect(source.validate_config!).to be_falsey
+    end
+
+    it 'identifies an invalid config when we fail to connect' do
+      expect(source).to receive(:names_to_ids).
+        and_raise(Excon::Errors::Unauthorized.new('test_error'))
+
+      expect(source.validate_config!).to be_falsey
+    end
+  end
+
+  context '#server_is_proxy_type?' do
+    it 'identifies a server that runs a proxy' do
+      server = double(
+        name: '7-5-6-4-12345-1234567890'
+      )
+
+      expect(source.send(:server_is_proxy_type?, server)).to be_truthy
+    end
+
+    it 'identifies a server that is not configured as expected' do
+      server = double(
+        name: '-5-6-4-12345-1234567890'
```

Comment:
A better test would be if the regex matched the name string but the values were different.  This would simulate a server created on the same account in a different region.